### PR TITLE
Adding stats and contributors to whatsnew

### PR DESCRIPTION
**Pull request by bsipocz**

### CODE REVIEW

File: docs/whatsnew/3.1.rst [hunk 1]
```diff
@@ -33,9 +33,11 @@ In addition to these major changes, Astropy 3.1 includes a large number of
 smaller improvements and bug fixes, which are described in the
 :ref:`changelog`. By the numbers:
 
-* xxx issues have been closed since v3.0
-* xxx pull requests have been merged since v3.0
-* xxx distinct people have contributed code
+
```
Comment (pllim) [inline docs/whatsnew/3.1.rst]:
```suggestion
```

File: docs/whatsnew/3.1.rst [hunk 2]
```diff
@@ -520,5 +522,105 @@ To see a detailed list of all changes in version v3.1, including changes in
 API, please see the :ref:`changelog`.
 
 
-Renamed/removed functionality
-=============================
+Contributors to the v3.1 release
+================================
+
+.. hlist::
```
Comment (pllim) [inline docs/whatsnew/3.1.rst]: Need to check rendered doc in CircleCI to make sure it renders as expected, as the GitHub preview does not.

File: docs/whatsnew/3.1.rst [hunk 3]
```diff
@@ -520,5 +522,105 @@ To see a detailed list of all changes in version v3.1, including changes in
...
+  *  Mike Alexandersen  *
+  *  Mikołaj  *
+  *  Nadia Dencheva
+  *  Ole Streicher
+  *  Patricio Rojo  *
+  *  Pey Lian Lim
+  *  Ritiek Malhotra  *
+  *  Rocio Kiman  *
+  *  Rohan Rajpal  *
+  *  SG004  *
```
Comment (pllim) [inline docs/whatsnew/3.1.rst]: Joseph Schlitz https://github.com/SG004

File: docs/whatsnew/3.1.rst (hunk 2 above)
Comment (bsipocz) [reply docs/whatsnew/3.1.rst]: Absolutely. Otherwise I works for me ™️ 😄

**Comment by bsipocz** [comment]
> Kudos to adding the name list!

Kudos to @Cadair for helping with the mailmap and extending the procedure script (that we need to add to our procedures, ideally in a PR by him :) )

**Comment by pllim** [comment]
> we need to add to our procedures, ideally in a PR by him

Should I open an issue? 😉

**Comment by pllim** [comment]
Hmm, CircleCI failure. @astrofrog ?
```
    Complete output from command python setup.py egg_info:
    Beginning with Matplotlib 3.1, Python 3.6 or above is required.
    This may be due to an out of date pip.
    Make sure you have pip >= 9.0.1.
```

File: docs/whatsnew/3.1.rst [hunk 4]
```diff
@@ -33,9 +33,11 @@ In addition to these major changes, Astropy 3.1 includes a large number of
 smaller improvements and bug fixes, which are described in the
 :ref:`changelog`. By the numbers:
 
-* xxx issues have been closed since v3.0
-* xxx pull requests have been merged since v3.0
-* xxx distinct people have contributed code
+
+* 906 issues have been closed since v3.0
```
Comment (eteq) [inline docs/whatsnew/3.1.rst]: 😮 (i.e., a big number!)

File: docs/whatsnew/3.1.rst [hunk 5]
```diff
@@ -33,9 +33,11 @@ In addition to these major changes, Astropy 3.1 includes a large number of
...
+* 437 pull requests have been merged since v3.0
+* 94 distinct people have contributed code to this release, 51 of which are
+  first time contributors to Astropy.
```
Comment (eteq) [inline docs/whatsnew/3.1.rst]: 🎉

**Comment by bsipocz** [comment]
> Hmm, CircleCI failure. @astrofrog ?

I would ignore it for now and go ahead with the release.

**Comment by eteq** [comment]
checking the build locally shortly... but the circleci failure appeared on #8222 in just the last few hours, so I agree it's probably ignorable for this PR (it looks like an mpl update happened in the meantime, so we probably just need to pin a version or something).

**Comment by bsipocz** [comment]
@eteq - you can look at giles, no need to build locally.

**Comment by pllim** [comment]
I am confused. I don't see the name list at https://circleci.com/gh/astropy/astropy/18377/artifacts/0//home/circleci/project/docs/_build/html/index.html

**Comment by bsipocz** [comment]
@pllim - https://18377-2081289-gh.circle-artifacts.com/0/home/circleci/project/docs/_build/html/whatsnew/3.1.html

**Comment by pllim** [comment]
Oh wait, I found it -- https://18377-2081289-gh.circle-artifacts.com/0/home/circleci/project/docs/_build/html/whatsnew/3.1.html but the index points to 3.2 🤷‍♀️ 

The list rendering is cool! 👍

**Comment by bsipocz** [comment]
@pllim - in the backport 3.1 what's new will be on the index page.

File: docs/whatsnew/3.1.rst (hunk 2 above)
Comment (eteq) [reply docs/whatsnew/3.1.rst]: I double checked this and it rendered fine locally!

**Pull request merged**